### PR TITLE
Add Functions::RayleighKotheVortex

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: source/base/function_lib.cc [hunk 1]
```diff
@@ -2947,6 +2947,34 @@ namespace Functions
...
+  template <int dim>
+  void
+  RayleighKotheVortex<dim>::vector_value(const Point<dim> &point,
+                                         Vector<double> &  values) const
+  {
+    const double t = this->get_time();
+
+    const double px = numbers::PI * point(0);
+    const double py = numbers::PI * point(1);
+    const double pt = numbers::PI / T * t;
```
Comment (blaisb) [inline source/base/function_lib.cc:2968]: Not sure if that's the best variable name. pt for me mentally translates to point... but maybe that's just me :)

File: examples/step-68/step-68.cc [hunk 2]
```diff
@@ -278,7 +236,7 @@ namespace Step68
     MappingQ1<dim>                             mapping;
     LinearAlgebra::distributed::Vector<double> velocity_field;
 
-    Vortex<dim> velocity;
+    Functions::RayleighKotheVortex<dim> velocity;
```
Comment (blaisb) [inline examples/step-68/step-68.cc:239]: Can you also make the same change in the performance benchmark while we are at it?
:)

**Review by blaisb — CHANGES_REQUESTED** [review]
Two small comments,  nothing big

File: source/base/function_lib.cc (hunk 1 above)
Comment (peterrum) [reply source/base/function_lib.cc:2968]: I am fine with the names.

**Review by blaisb — APPROVED** [review]

File: include/deal.II/base/function_lib.h [hunk 3]
```diff
@@ -1752,6 +1752,72 @@ namespace Functions
     const std::vector<double> coefficients;
   };
 
+
+  /**
+   * A class that represents a time-dependent function object for a
+   * Rayleigh--Kothe vortex vector field. This is generally used as
+   * flow pattern in complex test cases for interface tracking methods
```
Comment (masterleinad) [inline include/deal.II/base/function_lib.h:1759]:
```suggestion
   * as flow pattern in complex test cases for interface tracking methods
```
?

File: source/base/function_lib.cc (hunk 1 above)
Comment (masterleinad) [reply source/base/function_lib.cc:2968]: Maybe
```suggestion
    const double pi_t = numbers::PI / T * t;
```
and similarly above?

File: include/deal.II/base/function_lib.h (hunk 3 above)
Comment (peterrum) [reply include/deal.II/base/function_lib.h:1759]: The line above contains the `as`.

File: source/base/function_lib.cc (hunk 1 above)
Comment (peterrum) [reply source/base/function_lib.cc]: @masterleinad I went with your name suggestion. Thanks :+1:

**Pull request merged**